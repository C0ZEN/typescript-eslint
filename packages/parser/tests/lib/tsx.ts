import { parseForESLint } from '../../src/parser';
import { serializer } from '../tools/ts-error-serializer';

//------------------------------------------------------------------------------
// Tests
//------------------------------------------------------------------------------

expect.addSnapshotSerializer(serializer);

describe('TSX', () => {
  describe("if the filename ends with '.tsx', enable jsx option automatically.", () => {
    it('filePath was not provided', () => {
      const code = 'const element = <T/>';
      try {
        parseForESLint(code);
      } catch (e) {
        expect(e).toMatchSnapshot();
      }
    });

    it("filePath was not provided and 'jsx:true' option", () => {
      const code = 'const element = <T/>';
      expect(() =>
        parseForESLint(code, {
          ecmaFeatures: {
            jsx: true,
          },
        }),
      ).not.toThrow();
    });

    it('test.ts', () => {
      const code = 'const element = <T/>';
      try {
        parseForESLint(code, {
          filePath: 'test.ts',
        });
      } catch (e) {
        expect(e).toMatchSnapshot();
      }
    });

    it("test.ts with 'jsx:true' option", () => {
      const code = 'const element = <T/>';

      try {
        parseForESLint(code, {
          filePath: 'test.ts',
          ecmaFeatures: {
            jsx: true,
          },
        });
      } catch (e) {
        expect(e).toMatchSnapshot();
      }
    });

    it('test.tsx', () => {
      const code = 'const element = <T/>';
      expect(() =>
        parseForESLint(code, {
          filePath: 'test.tsx',
        }),
      ).not.toThrow();
    });

    it("test.tsx with 'jsx:false' option", () => {
      const code = 'const element = <T/>';
      expect(() =>
        parseForESLint(code, {
          ecmaFeatures: {
            jsx: false,
          },
          filePath: 'test.tsx',
        }),
      ).not.toThrow();
    });
  });
});
