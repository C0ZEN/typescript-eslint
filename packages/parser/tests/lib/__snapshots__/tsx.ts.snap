// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TSX if the filename ends with '.tsx', enable jsx option automatically. filePath was not provided 1`] = `
TSError {
  "column": 18,
  "index": 18,
  "lineNumber": 1,
  "message": "'>' expected.",
}
`;

exports[`TSX if the filename ends with '.tsx', enable jsx option automatically. test.ts 1`] = `
TSError {
  "column": 18,
  "index": 18,
  "lineNumber": 1,
  "message": "'>' expected.",
}
`;

exports[`TSX if the filename ends with '.tsx', enable jsx option automatically. test.ts with 'jsx:true' option 1`] = `
TSError {
  "column": 18,
  "index": 18,
  "lineNumber": 1,
  "message": "'>' expected.",
}
`;
